andela/ah-django
# #164859748 Send Registration Email

#### What does this PR do?
- Send an account verification email to a newly registered user

#### Description of Task to be completed?
 - [X] Enable registered users to verify the email account used for sign-up
- [X] Configure mail settings

#### How should this be manually tested?
1. Access a clone copy of the repo from [here](https://github.com/andela/ah-django)
```shell
$ git clone https://github.com/andela/ah-django
```
2. Checkout the this feature branch
```shell
$ git checkout ft-signup-email-notification-164859748
```
3. Create a virtual environment and activate it
```shell
$ pip install virtualenv && virtualenv venv && source venv/bin/activate
```
4. Install the project dependencies
```shell
$ pip install -r requirements.txt
```
5. Use the `.env_sample` to create an `.env` file with environment variables then source the .env.
The `env_sample` details out the required environment vairables. Copy them and paste in an `env` file. Export the environment variables with the command:
```
$ source .env
```
- Since this feature primarily works on email, ensure to ***provide an email address you have access to*** so you get to the cool part

6. Using curl or postman, test the endpoint `api/v1/users/` by sending a post request
- Provide the post data in this format:
```shell
	{
	    "user":  {
			"username": "about",
			"email": "to.say@waitforit",
			"password": "wheeeeeee!!!"
		}
	}
          
```


#### Any background context you want to provide?
- If you need to use sendgrid-api, comment out the sendgrid `EMAIL_BACKEND` in settings

#### What are the relevant pivotal tracker stories?
[#164859748](https://www.pivotaltracker.com/story/show/164859748)

#### Screenshots
![Screenshot from 2019-03-27 03-59-19-01](https://user-images.githubusercontent.com/44059971/55051864-fe4fb180-5067-11e9-807b-4ecb1f4f9412.jpeg)
